src/meta/MetaServiceUtils.cpp

```diff
@@ -294,12 +295,21 @@ cpp2::ErrorCode MetaServiceUtils::alterColumnDefs(std::vector<nebula::cpp2::Colu
             }
             break;
         }
-        case cpp2::AlterSchemaOp::DROP :
+        case cpp2::AlterSchemaOp::DROP:
         {
+            auto colName = col.get_name();
             for (auto it = cols.begin(); it != cols.end(); ++it) {
-                if (col.get_name() == it->get_name()) {
-                    cols.erase(it);
-                    return cpp2::ErrorCode::SUCCEEDED;
+                if (colName == it->get_name()) {
+                    // Check if there is a TTL on the column to be deleted
+                    if (!prop.get_ttl_col() ||
+                        (prop.get_ttl_col() && (*prop.get_ttl_col() != colName))) {
```

Comment:
i'm a bit curious about *prop.get_ttl_col() != colName

Reply:
What do you want to know